### PR TITLE
PPU LLVM: Add relocation 5 for ADDIS

### PR DESCRIPTION
**Pull request by elad335**
* Add more u16 relocations (4, 5, 6) support for instructions which already supported atleast one of them inclusing ADDIS as mentioned in #7855.
* More strict infinite desync detection, 
* Fix sys_prx_start/stop_module in liblv2.sprx HLE, add missing result reset.

### CODE REVIEW

File: rpcs3/Emu/Cell/PPUTranslator.cpp [hunk 1]
```diff
@@ -1787,6 +1787,12 @@ void PPUTranslator::ADDIS(ppu_opcode_t op)
 {
 	Value* imm = m_ir->getInt64(op.simm16 << 16);
 
+	if (m_rel && m_rel->type == 5)
+	{
+		imm = m_ir->CreateShl(ZExt(ReadMemory(GetAddr(+2), GetType<u16>()), GetType<u64>()), 16);
+		m_rel = nullptr;
+	}
+
 	if (m_rel && m_rel->type == 6)
```
Comment (AniLeo) [inline rpcs3/Emu/Cell/PPUTranslator.cpp]: else if

Comment (elad335) [reply rpcs3/Emu/Cell/PPUTranslator.cpp]: Solved with a better solution.

Comment (AniLeo) [reply rpcs3/Emu/Cell/PPUTranslator.cpp]: type 5 on ADDIS didn't require ZExt instead of SExt?

Comment (elad335) [reply rpcs3/Emu/Cell/PPUTranslator.cpp]: was a copypasta typo.

**Comment by sampletext32** [comment]
I can see a lot of duplicating checks. Maybe they are worth extracting into a func?

**Comment by Nekotekina** [comment]
> I can see a lot of duplicating checks. Maybe they are worth extracting into a func?

Not really, not in this case. This is also "old style" internal API.

**Pull request merged**